web/src/stream_create.ts

```diff
@@ -423,21 +423,22 @@ function create_stream(): void {
             // The rest of the work is done via the subscribe event we will get
         },
         error(xhr): void {
-            const error_message = z.object({msg: z.string().optional()}).parse(xhr.responseJSON);
-            if (error_message?.msg?.includes("access")) {
-                // If we can't access the stream, we can safely
-                // assume it's a duplicate stream that we are not invited to.
-                //
-                // BUG: This check should be using error codes, not
-                // parsing the error string, so it works correctly
-                // with i18n.  And likely we should be reporting the
-                // error text directly rather than turning it into
-                // "Error creating channel"?
+            const error_code = z.object({code: z.string().optional()}).parse(xhr.responseJSON);
```

Comment:
Not sure of the types here.

Is `code` optional here? I guess we `code` is included for all error types and other places where we use similar technique of handling different codes differently also always expect code. We instead check if `xhr.responseJSON` is `undefined` or not. Would be good to see how we do this at other places to get an idea.